src/poetry/utils/authenticator.py

```diff
@@ -82,10 +82,25 @@ def http_credential_keys(self) -> list[str]:
     def get_http_credentials(
         self, password_manager: PasswordManager, username: str | None = None
     ) -> HTTPAuthCredential:
-        # try with the repository name via the password manager
-        credential = HTTPAuthCredential(
-            **(password_manager.get_http_auth(self.name) or {})
-        )
+        # try with the URL
+        if "@" in self.netloc:
```

Comment:
This logic already exists in `get_credentials_for_url()` and shouldn't need to be copied.